clients/src/main/java/org/apache/kafka/common/security/authenticator/LoginManager.java

```diff
@@ -75,15 +94,18 @@ private static String getServiceName(String loginContext, Map<String, ?> configs
      *
      * @param loginType the type of the login context, it should be SERVER for the broker and CLIENT for the clients
      *                  (i.e. consumer and producer)
+     * @param mechanism SASL mechanism 
      * @param configs configuration as key/value pairs
      */
-    public static final LoginManager acquireLoginManager(LoginType loginType, Map<String, ?> configs) throws IOException, LoginException {
+    public static final LoginManager acquireLoginManager(LoginType loginType, SaslMechanism mechanism, Map<String, ?> configs) throws IOException, LoginException {
         synchronized (LoginManager.class) {
             LoginManager loginManager = CACHED_INSTANCES.get(loginType);
             if (loginManager == null) {
-                loginManager = new LoginManager(loginType, configs);
+                loginManager = new LoginManager(loginType, mechanism, configs);
                 CACHED_INSTANCES.put(loginType, loginManager);
             }
+            if (!loginManager.mechanism.equals(mechanism))
+                throw new LoginException("Multiple SASL mechanisms cannot be used for the same login type");
```

Comment:
Added mechanism to CACHED_INSTANCES to enable multiple mechanisms to be supported.